controller/userController.js

```diff
@@ -114,52 +104,47 @@ const create = (req, res) => {
             user.salt = retVal.salt;
             user.passwordHash = retVal.passwordHash;
             // Add new user
-            user.save().then(() => {
-                user.passwordHash = null;
-                user.salt = null;
-                return res.status(HttpStatus.CREATED).json(user);
+            user.save().then((newUser) => {
+                newUser.passwordHash = null;
+                newUser.salt = null;
+                return res.status(HttpStatus.CREATED).json(newUser);
             });
         } else {
             return res.status(HttpStatus.BAD_REQUEST).json({
-                status: StatusEnum['ERROR'],
                 message: 'User with provided email already exists!'
             });
         }
     }).catch(err => {
         return res.status(HttpStatus.BAD_REQUEST).json({
-            status: StatusEnum['ERROR'],
             message: ValidationUtil.buildErrorMessage(err, 'create', 'user')
         });
     });
 };
 
-const retrieveAll = (req, res) => {
+const getAll = (req, res) => {
     const user = req.decoded;
 
     User.find({ companyID: user.companyID }, { passwordHash: 0, salt: 0 }).then(users => {
         if (users) {
             return res.status(HttpStatus.OK).json(users);
         } else {
             return res.status(HttpStatus.NOT_FOUND).json({
-                status: StatusEnum['ERROR'],
                 message: 'Users not found.'
             });
         }
     }).catch(err => {
         return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json({
-            status: StatusEnum['ERROR'],
-            message: ValidationUtil.buildErrorMessage(err, 'retrieveAll', 'user')
+            message: ValidationUtil.buildErrorMessage(err, 'getAll', 'user')
         });
     });
 };
 
-const retrieveById = (req, res) => {
+const getById = (req, res) => {
     const userId = req.params.id;
     const loggedUser = req.decoded;
 
-    if (userId.length != AppConfig.OBJECT_ID_LEN) {
+    if (userId.length != AppConfig.OBJECT_ID_LEN || !req.params || !req.params.id) {
```

Comment:
No need to check for `req.params` since it is always present. Also you could have used the `ValidationUtil.isValidObjectId` function here.